### PR TITLE
Validate: (data) Block can't appear within a Block

### PR DESCRIPTION
**Pull request by sarahM0**
To fix: https://github.com/KhronosGroup/SPIRV-Tools/issues/1587
Please review.

### CODE REVIEW

File: source/val/validate_instruction.cpp [hunk 1]
```diff
@@ -379,6 +385,14 @@ spv_result_t LimitCheckStruct(ValidationState_t& _, const Instruction* inst) {
            << "Structure Nesting Depth may not be larger than " << depth_limit
            << ". Found " << cur_depth << ".";
   }
+
+  _.SetHasNestedBlockOrBufferBlockStruct(inst->id(), has_nested_blockOrBufferBlock_struct);
+  if(_.HasNestedBlockOrBufferBlockStruct(inst->id()) && _.HasDecoration(inst->id(), SpvDecorationBlock)){
```
Comment (sarahM0) [inline source/val/validate_instruction.cpp]: This always returns false. Could tell me what am I doing wrong?

Comment (alan-baker) [reply source/val/validate_instruction.cpp]: I think the decorations have not been registered at this point. Your check should be performed later in the flow.

File: test/val/val_decoration_test.cpp [hunk 2]
```diff
@@ -2303,18 +2341,18 @@ TEST_F(ValidateDecorations, MultiplePushConstantsSingleEntryPointGood) {
                 OpMemoryModel Logical GLSL450
                 OpEntryPoint Fragment %1 "main"
                 OpExecutionMode %1 OriginUpperLeft
-
```
Comment (sarahM0) [inline test/val/val_decoration_test.cpp]: Do I need to be worried about these auto indentations which are done by my editor(Sublime)?

Comment (alan-baker) [reply test/val/val_decoration_test.cpp]: Not as long as the tests keep passing and the format checker is ok.

File: source/val/validate_instruction.cpp (hunk 1 above)
Comment (sarahM0) [reply source/val/validate_instruction.cpp]: That might be it. In that case I think we need to update  _HasDecoration_ definition to return an error, saying it doesn't know yet. Also I couldn't fully understand how this function works. I appreciate it if you could go through it with me when you were in office.

File: source/opt/block_merge_util.cpp [hunk 3]
```diff
@@ -86,6 +86,11 @@ bool CanMergeWithSuccessor(IRContext* context, BasicBlock* block) {
     return false;
   }
 
+  // Don't bother trying to merge unreachable blocks.
```
Comment (alan-baker) [inline source/opt/block_merge_util.cpp]: This looks like a bad rebase.

**Comment by sarahM0** [comment]
closed because of a bad rebase.

**Pull request closed without merging**